core/crypto/src/vrf.rs

```diff
@@ -0,0 +1,354 @@
+use blake2::{Blake2b, VarBlake2b};
+use bs58;
+use curve25519_dalek::constants::{RISTRETTO_BASEPOINT_POINT as G, RISTRETTO_BASEPOINT_TABLE as GT};
```

Comment:
we also use sodiumoxide for 25519 curve. Does it not support the required operations?

Reply:
Unfortunately libsodium doesn't provide low-level operations necessary to implement a VRF in its public API.